perun-wui-profile/src/main/java/cz/metacentrum/perun/wui/profile/client/PerunProfileUtils.java

```diff
@@ -73,8 +84,8 @@ public static Integer getUserId(PlaceManager placeManager) {
 	 * @return true if given value is correct ssh key, false otherwise
 	 */
 	public static boolean isValidSshKey(String value) {
-		for (String prefix : PerunProfileUtils.SSH_KEY_PREFIXES) {
-			if (value.startsWith(prefix)) {
+		for (String type : PerunProfileUtils.SSH_KEY_TYPES) {
+			if (value.trim().equals(type.trim())) {
```

Comment:
I think `trim()` returns string without leading/ending whitespace and not substring from start to first whitespace. We pass whole SSH key value in this method, so you should still check it as a prefix and not as equals.